### DISCUSSION TITLE
How to write a `.scm` script to configure the treesitter?

### DISCUSSION THREAD

**Discussion by nullptr-yxw** [nvim-treesitter/nvim-treesitter#6004] [Q&A] [answered]
In `:h treesitter`, there are many scheme script to configure the treesitter. However, I don't know where and how can I write these scheme script.

**Comment by clason** [comment]
https://tree-sitter.github.io/tree-sitter/syntax-highlighting#queries

**Comment by nullptr-yxw** [comment]
@clason So where should I write this script?

**Reply by clason** [reply, marked as answer]
Anywhere in `runtimepath`, under `queries/<language>/<type>.scm`. See `:h treesitter-query`, third paragraph.

**Reply by nullptr-yxw** [reply]
@clason Is it possible for a colorscheme plugin to do this? I want to create a `PR` to a colorscheme.

**Reply by clason** [reply]
No. Colorschemes have no business bundling queries, this will only lead to problems. (Believe me, we've had _very_ bad experiences with this in the past.)
